molecule/testinfra/staging/app-code/test_securedrop_rqrequeue.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+
+testinfra_hosts = ["app-staging"]
+
+
+def test_securedrop_rqrequeue_service(host):
```

Comment:
I'm still testing at the moment but I'm realizing that rqrequeue will also no longer re-enqueue hashing jobs that aren't completed, i.e. it's not going to work as advertised as the `StartedJobRegistry` assumption it's based upon is no longer valid (we know those jobs will be in the failed job registry but we can't tell if they failed due to interruption during restart or some other reason).

My thinking is that we'll want to address that by just removing the `rqrequeue` service entirely - this one is easier to handle from the application's perspective as we can just on app start resubmit jobs for any files that don't have a hash stored in the database. Anyway since this is an even larger change and we want to get _this_ change in for 1.2.0, let's do in a followup. Let me know if you think that approach doesn't make sense.

Reply:
It's affected the same way by the switch to systemd, yes. But it's less of a problem, as we're just using the checksums for ETags, and if they're missing they're [calculated as needed](https://github.com/freedomofpress/securedrop/blob/ed014bed5f810b71f86c4bd556a6c73679587f53/securedrop/journalist_app/utils.py#L346).

My vote is to move hashing to another simple systemd service and remove rq entirely. It's too late to do it here, but we'd eliminate a lot of code and config, at least rq as a dependency we need to keep updated and reviewed, and possibly also Redis?

Reply:
oh yeah they're calculated on the fly, great point. Since we need to serve those in the headers as responses from the API, we should ensure they are hashed async prior to the user requesting them as at least for large files it takes too long (`file.out` is 500 MB here):

```
www-data@app-staging:/var/www/securedrop$ time shasum -a 256 file.out
38f7c0648553d81ad9402ebdd1b275a0029644c5b7eef7c963dfa7db9ef0ba23  file.out

real    0m3.278s
user    0m2.557s
sys     0m0.452s
```

We could continue to use rq or replace rqrequeue with another systemd service just for hashing (open to either tbh), to discuss further and then incorporate as part of 1.3.0 🚀